### ISSUE TITLE
What format to use for semantic segmentation labels?

### ISSUE THREAD

**Issue by chauvenne** [facebookresearch/detectron2#2376]
## ❓ What format to use for semantic segmentation labels?

Hello, first of all I would like to thank you for the work done on detectron2 :)

I'm trying to train a semantic segmentation model (DeepLab) on a custom dataset. The dataset is composed of multispectral images (i.e. more than the 3 RGB channels) and thus I've written my own DataLoader. It seems to work.

The goal of the model is to segment clouds, so I've only one class. The other pixels should be considered as background.
The pixels in the ground truth label masks are as follows: 

- 0 is background
- 1 is cloud

In my training script, I've set the following metadata: 
`MetadataCatalog.get(DATASET_NAME).set(
            stuff_classes=["background", "cloud"],
            evaluator_type="sem_seg",
        )`
The number of classes in the config file was set to 2.

Now, the model is training but the loss is not decreasing. 

Is it the correct way to format the labels? Or should I set the `ignore_labels` to 0 and use only one class ?

**Comment by ppwwyyxx** [comment]
> Is it the correct way to format the labels

yes

> Or should I set the ignore_labels to 0 and use only one class ?

it's up to you whether to ignore background labels or not.
